server/mux.go

```diff
@@ -67,127 +67,137 @@ func NewMux(opts MuxOpts, service Service) http.Handler {
 		hr.NotFound = http.StripPrefix(opts.Basepath, hr.NotFound)
 	}
 
+	EnsureViewer := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, ViewerRoleName, opts.Logger, next)
+	}
+	EnsureEditor := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, EditorRoleName, opts.Logger, next)
+	}
+	EnsureAdmin := func(next http.HandlerFunc) http.HandlerFunc {
+		return AuthorizedUser(service.UsersStore, opts.UseAuth, AdminRoleName, opts.Logger, next)
+	}
+
 	/* Documentation */
 	router.GET("/swagger.json", Spec())
 	router.GET("/docs", Redoc("/swagger.json"))
 
 	/* API */
 	// Sources
-	router.GET("/chronograf/v1/sources", service.Sources)
-	router.POST("/chronograf/v1/sources", service.NewSource)
+	router.GET("/chronograf/v1/sources", EnsureViewer(service.Sources))
+	router.POST("/chronograf/v1/sources", EnsureEditor(service.NewSource))
 
-	router.GET("/chronograf/v1/sources/:id", service.SourcesID)
-	router.PATCH("/chronograf/v1/sources/:id", service.UpdateSource)
-	router.DELETE("/chronograf/v1/sources/:id", service.RemoveSource)
+	router.GET("/chronograf/v1/sources/:id", EnsureViewer(service.SourcesID))
+	router.PATCH("/chronograf/v1/sources/:id", EnsureEditor(service.UpdateSource))
+	router.DELETE("/chronograf/v1/sources/:id", EnsureEditor(service.RemoveSource))
 
 	// Source Proxy to Influx; Has gzip compression around the handler
 	influx := gziphandler.GzipHandler(http.HandlerFunc(service.Influx))
 	router.Handler("POST", "/chronograf/v1/sources/:id/proxy", influx)
 
 	// Write proxies line protocol write requests to InfluxDB
-	router.POST("/chronograf/v1/sources/:id/write", service.Write)
+	router.POST("/chronograf/v1/sources/:id/write", EnsureEditor(service.Write))
 
 	// Queries is used to analyze a specific queries
-	router.POST("/chronograf/v1/sources/:id/queries", service.Queries)
+	router.POST("/chronograf/v1/sources/:id/queries", EnsureEditor(service.Queries))
 
 	// All possible permissions for users in this source
-	router.GET("/chronograf/v1/sources/:id/permissions", service.Permissions)
+	router.GET("/chronograf/v1/sources/:id/permissions", EnsureAdmin(service.Permissions))
```

Comment:
This should be EnsureViewer

Permissions describe the types of permissions a data source provides.